v0/destinations/mautic/transform.js

```diff
@@ -0,0 +1,247 @@
+    message,
+    mappingConfig[ConfigCategories.IDENTIFY.name]
+  );
+  if (message.traits.address || message.context.traits.address) {
+    let add;
+    if (message.traits.address) {
+      add = message.traits.address;
+    } else {
+      add = message.context.traits.address;
+    }
+    const len = add.length;
+    if (len > 128) {
+      throw new CustomError(
+        "Please provide an Address with less than 128 char ",
+        400
+      );
+    }
+    set(payload, "address1", add.substr(0, len / 2));
+    set(payload, "address2", add.substr(len / 2 + 1, len - 1));
+  }
+  console.log("Payload:", payload);
+  const endpoint = `${BASE_URL}/contacts/new`;
+  return responseBuilderIdentify(
+    payload,
+    endpoint,
+    ConfigCategories.IDENTIFY.method,
+    destination
+  );
+};
+// Includes the fields that can be lookUpfields
+const Fields = [
+  "",
+  "id",
+  "title",
+  "role",
+  "firstname",
+  "nps__recommend",
+  "lastname",
+  "cart_status",
+  "company",
+  "sandbox",
+  "car_or_truck",
+  "position",
+  "email",
+  "company_size",
+  "mobile",
+  "prospect_or_customer",
+  "datetime",
+  "phone",
+  "points",
+  "subscription_status",
+  "fax",
+  "b2b_or_b2c",
+  "products",
+  "address1",
+  "address2",
+  "haspurchased",
+  "city",
+  "crm_id",
+  "state",
+  "zipcode",
+  "country",
+  "preferred_locale",
+  "timezone",
+  "last_active",
+  "attribution_date",
+  "attribution",
+  "website",
+  "facebook",
+  "foursquare",
+  "instagram",
+  "linkedin",
+  "skype",
+  "twitter"
+];
+const process = async event => {
+  // console.log(event);
+  const { message, destination } = event;
+  const { Config } = destination;
+  // checking for the configurations validations
+  if (!Fields.includes(Config.lookUpField)) {
+    throw new CustomError("Invalid Lookup Field", 400);
+  }
+  if (!Config.password) {
+    throw new CustomError("Password field can not be empty", 400);
+  }
+  if (!Config.subDomainName) {
+    throw new CustomError("Sub-Domain Name field can not be empty", 400);
+  }
+  if (!ValidateEmail(Config.userName)) {
+    throw new CustomError("User Name is not Valid", 400);
+  }
+
+  // Validating if message type is even given or not
+  if (!message.type) {
+    throw new CustomError(
+      "Message Type is not present. Aborting message.",
+      400
+    );
+  }
+  const messageType = message.type.toLowerCase();
+  let response;
+  // Switching to particular Message Event
+  BASE_URL = BASEURL.replace("subDomainName", Config.subDomainName);
+
+  switch (messageType) {
+    case EventType.IDENTIFY:
+      response = await identifyResponseBuilder(message, destination);
+      break;
+    case EventType.GROUP:
+      response = await groupResponseBuilder(message, destination);
+      break;
+    default:
+      throw new CustomError(`Message type ${messageType} not supported`, 400);
+  }
+  console.log("Response: ",response);
+  return response;
+};
+module.exports = { process };
```

Comment:
Add router transform function over here. And add router test cases for the same. As we have axios call for both identify and group call, we would need to mock the test cases as well. @Gauravudia can you please help regarding this?